src/fetch/orderbook.py

```diff
@@ -0,0 +1,251 @@
+    """
+
+    @staticmethod
+    def pg_engine(db_env: OrderbookEnv) -> Engine:
+        """Returns a connection to postgres database"""
+        load_dotenv()
+        if db_env == OrderbookEnv.ANALYTICS:
+            db_url = os.environ["ANALYTICS_DB_URL"]
+            db_string = f"postgresql+psycopg2://{db_url}"
+            return create_engine(db_string)
+
+        db_url = os.environ[f"{db_env}_DB_URL"]
+        db_string = f"postgresql+psycopg2://{db_url}"
+        return create_engine(db_string)
+
+    @classmethod
+    def _read_query_for_env(
+        cls, query: str, env: OrderbookEnv, data_types: Optional[dict[str, str]] = None
+    ) -> DataFrame:
+        return pd.read_sql_query(query, con=cls.pg_engine(env), dtype=data_types)
+
+    @classmethod
+    def _query_both_dbs(
+        cls,
+        query_prod: str,
+        query_barn: str,
+        data_types: Optional[dict[str, str]] = None,
+    ) -> tuple[DataFrame, DataFrame]:
+        barn = cls._read_query_for_env(query_barn, OrderbookEnv.BARN, data_types)
+        prod = cls._read_query_for_env(query_prod, OrderbookEnv.PROD, data_types)
+        return barn, prod
+
+    @classmethod
+    def get_order_rewards(cls, block_range: BlockRange) -> DataFrame:
+        """
+        Fetches and validates Order Reward DataFrame as concatenation from Prod and Staging DB
+        """
+        cow_reward_query_prod = (
+            open_query("orderbook/prod_order_rewards.sql")
+            .replace("{{start_block}}", str(block_range.block_from))
+            .replace("{{end_block}}", str(block_range.block_to))
+        )
+        cow_reward_query_barn = (
+            open_query("orderbook/barn_order_rewards.sql")
+            .replace("{{start_block}}", str(block_range.block_from))
+            .replace("{{end_block}}", str(block_range.block_to))
+        )
+        data_types = {"block_number": "int64", "amount": "float64"}
+        barn, prod = cls._query_both_dbs(
+            cow_reward_query_prod, cow_reward_query_barn, data_types
+        )
+
+        # Warn if solver appear in both environments.
+        if not set(prod.solver).isdisjoint(set(barn.solver)):
+            log.warning(
+                f"solver overlap in {block_range}: solvers "
+                f"{set(prod.solver).intersection(set(barn.solver))} part of both prod and barn"
+            )
+
+        if not prod.empty and not barn.empty:
+            return pd.concat([prod, barn])
+        if not prod.empty:
+            return prod.copy()
+        if not barn.empty:
+            return barn.copy()
+        return pd.DataFrame()
+
+    @classmethod
+    def run_batch_data_query(cls, block_range: BlockRange) -> DataFrame:
+        """
+        Fetches and validates Batch data DataFrame as concatenation from Prod and Staging DB
+        """
+        load_dotenv()
+        network = node_suffix(os.environ["NETWORK"])
+        epsilon_upper = str(os.environ[f"EPSILON_UPPER_{network}"])
```

Comment:
If we really want to merge these code bases, this should probably use the config object.